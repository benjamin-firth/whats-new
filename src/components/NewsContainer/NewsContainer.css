.articleSection {
  display: flex;
  flex-wrap: wrap;
  height: 100vh;
  justify-content: center;
  width: 80%;
}