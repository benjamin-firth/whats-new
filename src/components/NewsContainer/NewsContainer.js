import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = ({ props }) => {
  const newsArticles = props.map(article => {
    return <NewsArticle 
      img={article.img}
      headline={article.headline}
      description={article.description}
      url={article.url}
      key={article.id}
    />
  })

  return (
    <main className="articleSection">
      {newsArticles}
    </main>
  )
}

export default NewsContainer;
