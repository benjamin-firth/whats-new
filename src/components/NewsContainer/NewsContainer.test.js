import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  const mockProps = [{
    img:'image',
    headline:'headline',
    description:'my mock description',
    url:'www.google.com',
    id:'1234'
  }]

  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsContainer 
      props={mockProps}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})