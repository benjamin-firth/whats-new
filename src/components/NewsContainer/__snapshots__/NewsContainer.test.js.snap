// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<main
  className="articleSection"
>
  <NewsArticle
    description="my mock description"
    headline="headline"
    img="image"
    key="1234"
    url="www.google.com"
  />
</main>
`;
